src/libs/rover/rover.cpp

```diff
@@ -210,15 +211,170 @@ class Rover::InternalsType
 
 }; //Internals Type
 
-Rover::Rover()
-  : m_internals( new InternalsType )
+Rover::Rover() : m_internals( new InternalsType )
 {
 
 }
 
 Rover::~Rover()
 {
+#ifdef ROVER_ENABLE_LOGGING
+  DataLogger::GetInstance()->WriteLog();
+#endif
+}
+
+void
+Rover::update_settings(Node &params)
+{
+  if (params.has_path("camera/look_at"))
+  {
+    m_settings["camera/look_at"].set(params["camera/look_at"]);
+  }
+
+  if (params.has_path("camera/up"))
+  {
+    m_settings["camera/up"].set(params["camera/up"]);
+  }
+
+  if (params.has_path("camera/fov"))
+  {
+    m_settings["camera/fov"] = params["camera/fov"];
+  }
+
+  if (params.has_path("camera/xpan"))
+  {
+    m_settings["camera/xpan"] = params["camera/xpan"];
+  }
+
+  if (params.has_path("camera/ypan"))
+  {
+    m_settings["camera/ypan"] = params["camera/ypan"];
+  }
 
+  if(params.has_path("camera/zoom"))
+  {
+    m_settings["camera/zoom"] = params["camera/zoom"];
+  }
+
+  if(params.has_path("camera/near_plane"))
+  {
+    m_settings["camera/near_plane"] = params["camera/near_plane"];
+  }
+
+  if (params.has_path("camera/far_plane"))
+  {
+    m_settings["camera/far_plane"] = params["camera/far_plane"];
+  }
+
+  if (params.has_path("camera/image_width"))
+  {
+    m_settings["camera/image_width"] = params["camera/image_width"]; 
+  }
+
+  if (params.has_path("camera/image_height"))
+  {
+    m_settings["camera/image_height"] = params["camera/image_height"]; 
+  }
+}
+
+void
+Rover::print_settings()
+{
+  std::cout << m_settings.to_yaml() << std::endl;
+}
+
+void
+Rover::update_camera()
+{
+  if (m_settings.has_path("camera/look_at"))
+  {
+    float64_accessor focus = m_settings["camera/look_at"].value();
+    vtkmVec3f look_at(focus[0], focus[1], focus[2]);
+    m_camera.SetLookAt(look_at);
+  }
+  
+  if (m_settings.has_path("camera/up"))
+  {
+    float64_accessor view_up = m_settings["camera/up"].value();
+    vtkmVec3f up(view_up[0], view_up[1], view_up[2]);
+    m_camera.SetViewUp(up);
+  }
+  
+  if (m_settings.has_path("camera/fov"))
+  {
+    float64 view_angle = m_settings["camera/fov"].value();
+    m_camera.SetFieldOfView(view_angle);
```

Comment:
Let's have consistent names
```suggestion
    float64 fov = m_settings["camera/fov"].value();
    m_camera.SetFieldOfView(fov);
```

Reply:
Good catch.